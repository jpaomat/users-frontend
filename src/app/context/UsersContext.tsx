import { createContext } from 'react';
import { UsersState } from '../interfaces';

export type UserContextProps = {
    usersState: UsersState;
    setNumberPosts: ( numPosts: number ) => void;
    setNumberAlbums: ( numAlbums: number ) => void;
    setUserSelected: ( user: string ) => void;
} 

export const UserContext = createContext<UserContextProps>({} as UserContextProps);
