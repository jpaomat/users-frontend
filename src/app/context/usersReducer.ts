import { UsersState } from '../interfaces';

type UsersAction =
    | { type: 'addNumberPosts', payload: number }
    | { type: 'addNumberAlbums', payload: number };

export const usersReducer = (state: UsersState, action: UsersAction): UsersState => {
    switch (action.type) {
        case 'addNumberPosts':
            return {
                ...state,
                numberPosts: action.payload
            };
        case 'addNumberAlbums':
            return {
                ...state,
                numberAlbums: action.payload
            };
        default:
            return state;
    }
};
