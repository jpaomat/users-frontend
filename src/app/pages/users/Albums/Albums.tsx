import { useAlbums } from '../../../hooks';
import { UserTemplate } from '../../../templates';
import envJson from '../../../config/ENV.json';

export const Albums = () => {
  const usersClient = envJson.usersClient;
  const { albumsData, isLoadingData } = useAlbums(usersClient.albums, 5);

	return <UserTemplate dataList={albumsData} isLoadingData={isLoadingData} />;
};
