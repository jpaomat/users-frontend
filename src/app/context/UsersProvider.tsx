import { useReducer, JSX } from 'react';
import { UserContext } from './UsersContext';
import { UsersState } from '../interfaces';
import { usersReducer } from './usersReducer';

const INITIAL_STATE: UsersState = {
  numberAlbums: 5,
  numberPosts: 5
};

interface UserProviderProps {
	children: JSX.Element | JSX.Element[];
}

export const UsersProvider = ({children}: UserProviderProps) => {
  const [usersState, dispatch] = useReducer( usersReducer,INITIAL_STATE);

  const setNumberPosts = (numPosts: number) => {
    dispatch({type: 'addNumberPosts', payload: numPosts});
  };

  const setNumberAlbums = (numAlbums: number) => {
    dispatch({type: 'addNumberAlbums', payload: numAlbums});
  };

  return (
    <UserContext.Provider value={{usersState, setNumberPosts, setNumberAlbums}}>
      {children}
    </UserContext.Provider>
  );
};
