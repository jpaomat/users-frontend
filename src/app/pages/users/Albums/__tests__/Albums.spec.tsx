import { render } from '@testing-library/react';
import { Albums } from '../Albums';
import { UsersProvider } from '../../../../context/UsersProvider';

describe('<Albums /> component', () => {
    it('should render component', () => {
		const {container} = render(
			<UsersProvider>
				<Albums />
			</UsersProvider>
		);
		expect(container).toMatchSnapshot();
	});
});
