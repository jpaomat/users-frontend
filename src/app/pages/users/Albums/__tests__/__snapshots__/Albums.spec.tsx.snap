// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Albums /> component should render component 1`] = `
<div>
  <div
    class="container mt-72"
  >
    <div
      class="user-list"
    >
      <ul
        class="list-group"
      >
        <li
          class="list-group-item"
        >
          <div
            class=" d-flex m-8 justify-content-between"
          >
            <div>
              <blockquote
                class="blockquote"
              >
                <p
                  class="font-16"
                />
              </blockquote>
              <figcaption
                class="blockquote-footer"
              >
                <cite
                  title="Source Title"
                />
              </figcaption>
            </div>
            <div
              class="item-info__aditional"
            >
              <p
                class="text-primary font-16"
              />
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
